src/x509.c

```diff
@@ -7565,6 +7565,45 @@ int wolfSSL_X509_REVOKED_get_serial_number(RevokedCert* rev,
     return WOLFSSL_SUCCESS;
 }
 
+/* Retrieve the revocation date from RevokedCert
+ * return WOLFSSL_SUCCESS on success
+ */
+int wolfSSL_X509_REVOKED_get_revocationDate(RevokedCert* rev,
```

Comment:
Where is this used? I don't see a \#define mapping it to an OpenSSL function.

Also, stick to one type of casing. Don't mix camelCase and snake_case. So change `wolfSSL_X509_REVOKED_get_revocationDate` to `wolfSSL_X509_REVOKED_get_revocation_date`, like we have `wolfSSL_X509_REVOKED_get_serial_number`.

Reply:
HaProxy uses:

`X509_REVOKED_get0_serialNumber() `and `X509_REVOKED_get0_revocationDate()`. 

Our `wolfSSL_X509_REVOKED_get_serial_number `does not have a #define mapping so I didn't add one for the new `wolfSSL_X509_REVOKED_get_revocation_date`. Also, the already existing `wolfSSL_X509_REVOKED_get_serial_number does not follow the API spec in OpenSSL. Params and return value is different. Maybe that is why it doesn't have a #define mapping? Or am I missing something?

Reply:
If HaProxy has `X509_REVOKED_get0_serialNumber` and `X509_REVOKED_get0_revocationDate`, we need to add support for both, which will require a #define for both and a `wolfSSL_` prefixed version of each. Without that, I wouldn't expect HaProxy to compile with wolfSSL, unless the code using these functions doesn't actually get built.

I'm not sure what the story is behind `wolfSSL_X509_REVOKED_get_serial_number`. Looks like @lealem47 added it. We can probably leverage it when you implement `wolfSSL_X509_REVOKED_get0_serialNumber` (i.e. `wolfSSL_X509_REVOKED_get0_serialNumber` may just end up being a wrapper around `wolfSSL_X509_REVOKED_get_serial_number`).

Reply:
It gets compiled. You just have to do extra work around the API for example in HaProxy:

```
        sz = sizeof(serial);

        XMEMSET(serial, 0, sz);
        if (X509_REVOKED_get_serial_number(rev_entry, serial, &sz)
                == WOLFSSL_SUCCESS) {
            i2a_ASN1_INTEGER(bio, (ASN1_INTEGER*)serial);
        }
        else
            goto end;
```
We have internal wolfSSL code that does this. I'll add the #defines (have already) but if preferred I can implement those two new functions in a seperate PR.

Reply:
@haydenroche5, you're correct that `wolfSSL_X509_REVOKED_get_serial_number` can be leveraged to implement an openssl compatible `wolfSSL_X509_REVOKED_get0_serialNumber`. @Uriah-wolfSSL the way you would do this is very similar to the snippet of code you put up above.

Reply:
I'll add those two new OpenSSL APIs.

Reply:
Looking at HaProxy 2.6.0, I see this

```
i2a_ASN1_INTEGER(bio, (ASN1_INTEGER*)X509_REVOKED_get0_serialNumber(rev_entry));
```

We want our patch to be as un-instrusive as possible, so replacing `X509_REVOKED_get0_serialNumber` with `X509_REVOKED_get_serial_number` isn't ideal. We should just map `X509_REVOKED_get0_serialNumber` to `wolfSSL_X509_REVOKED_get0_serialNumber` with a #define, as we do for all other compat layer functions. It may just be happy accident that this works already, as HaProxy has a header openssl-compat.h that implements several OpenSSL functions they need in the event the OpenSSL version being used doesn't provide them (e.g. if the user is building HaProxy against OpenSSL 1.0.2 but HaProxy needs some functions that were only introduced in 1.1.1).

Reply:
New OpenSSL compat API's to be added in a follow-up PR.